job-server/src/main/scala/spark/jobserver/AkkaClusterSupervisorActor.scala

```diff
@@ -124,20 +124,31 @@ class AkkaClusterSupervisorActor(daoActor: ActorRef, dataManagerActor: ActorRef)
           (contextFromDAO, callbacks) match {
             case (Some(JobDAOActor.ContextResponse(Some(contextInfo))),
                 Some((successCallback, failureCallback))) =>
-              val contextConfig = ConfigFactory.parseString(contextInfo.config)
-              val isAdhoc = contextConfig.getBoolean("is-adhoc")
-              initContext(contextConfig, actorName,
-                     actorRef, contextInitTimeout)(isAdhoc, successCallback, failureCallback)
+              initContext(contextInfo, actorName, actorRef, false)(successCallback, failureCallback)
             case (Some(JobDAOActor.ContextResponse(None)), _) =>
               logger.error(
                  s"No such contextId ${contextId} was found in DB. Cannot initialize actor ${actorName}")
               actorRef ! PoisonPill
             case (Some(JobDAOActor.ContextResponse(Some(contextInfo))), None) =>
-              val exception = NoCallbackFoundException(contextInfo.id, actorRef.path.toSerializationFormat)
-              logger.error(exception.getMessage, exception)
-              actorRef ! PoisonPill // Since no watch was added, Terminated will not be fired
-              daoActor ! JobDAOActor.SaveContextInfo(contextInfo.copy(
-                  state = ContextStatus.Error, endTime = Some(DateTime.now()), error = Some(exception)))
+              isSuperviseModeEnabled(config, ConfigFactory.parseString(contextInfo.config)) match {
+                case isRestartScenario @ true =>
```

Comment:
Wait, wouldn't `isRestartScenario` always be true here?

Reply:
Yes, it will always be true. I just added these conditions to make the restart scenario more tight (just in case, some unknown scenario happen).